pkg/util/ovs/fake_ovs.go

```diff
@@ -0,0 +1,447 @@
+
+const (
+	parseForAdd    parseCmd = "add-flow"
+	parseForDelete parseCmd = "del-flows"
+)
+
+func fieldSet(parsed *ovsFlow, field string) bool {
+	for _, f := range parsed.fields {
+		if f.name == field {
+			return true
+		}
+	}
+	return false
+}
+
+func checkNotAllowedField(flow string, parsed *ovsFlow, field string, cmd parseCmd) error {
+	if fieldSet(parsed, field) {
+		return fmt.Errorf("bad flow %q (field '%s' not allowed in %s)", flow, field, cmd)
+	}
+	return nil
+}
+
+func checkUnsupportedField(flow string, parsed *ovsFlow, field string) error {
+	if fieldSet(parsed, field) {
+		return fmt.Errorf("bad flow %q (field '%s' not supported)", flow, field)
+	}
+	return nil
+}
+
+func parseFlow(cmd parseCmd, flow string, args ...interface{}) (*ovsFlow, error) {
+	if len(args) > 0 {
+		flow = fmt.Sprintf(flow, args...)
+	}
+
+	// According to the man page, "flow descriptions comprise a series of field=value
+	// assignments, separated by commas or white space." However, you can also have
+	// fields with no value (eg, "ip"), and the "actions" field can also have internal
+	// commas, whitespace, and equals signs (but if it is present, it must be the
+	// last field specified).
+
+	parsed := &ovsFlow{
+		table:    0,
+		priority: defaultPriority,
+		fields:   make([]ovsField, 0),
+		created:  time.Now(),
+		cookie:   "0",
+	}
+	for flow[0] == ' ' {
+		flow = flow[1:]
+	}
+	origFlow := flow
+	for flow != "" {
+		field := ""
+		value := ""
+
+		end := strings.IndexAny(flow, ", ")
+		if end == -1 {
+			end = len(flow)
+		}
+		eq := strings.Index(flow, "=")
+		if eq == -1 || eq > end {
+			field = flow[:end]
+		} else {
+			field = flow[:eq]
+			if field == "actions" {
+				end = len(flow)
+				value = flow[eq+1:]
+			} else {
+				valueEnd := end - 1
+				for flow[valueEnd] == ' ' || flow[valueEnd] == ',' {
+					valueEnd--
+				}
+				value = flow[eq+1 : valueEnd+1]
+			}
+			if value == "" {
+				return nil, fmt.Errorf("bad flow definition %q (empty field %q)", origFlow, field)
+			}
+		}
+
+		switch field {
+		case "table":
+			table, err := strconv.Atoi(value)
+			if err != nil {
+				return nil, fmt.Errorf("bad flow %q (bad table number %q)", origFlow, value)
+			} else if table < 0 || table > 255 {
+				return nil, fmt.Errorf("bad flow %q (table number %q out of range)", origFlow, value)
+			}
+			parsed.table = table
+		case "priority":
+			priority, err := strconv.Atoi(value)
+			if err != nil {
+				return nil, fmt.Errorf("bad flow %q (bad priority %q)", origFlow, value)
+			} else if priority < minPriority || priority > maxPriority {
+				return nil, fmt.Errorf("bad flow %q (priority %q out of range)", origFlow, value)
+			}
+			parsed.priority = priority
+		case "actions":
+			parsed.actions = value
+		case "cookie":
+			parsed.cookie = value
+		default:
+			parsed.fields = append(parsed.fields, ovsField{field, value})
+		}
+
+		for end < len(flow) && (flow[end] == ',' || flow[end] == ' ') {
+			end++
+		}
+		flow = flow[end:]
+	}
```

Comment:
That's done below; it's mandatory when doing AddFlow but forbidden when doing DelFlows